packages/ui/src/composables/useTextColor.ts

```diff
@@ -0,0 +1,15 @@
+import { computed, getCurrentInstance, Ref, unref } from 'vue'
+import { useColors } from '../services/color-config/color-config'
+
+export const useTextColor = (componentColor: string | Ref<string>, isTransparent: boolean | Ref<boolean> = false) => {
+  const { props } = getCurrentInstance()!
+  const { getColor, getTextColor } = useColors()
+  const textColorComputed = computed(() => {
+    if (unref(isTransparent)) {
+      return getColor(unref(componentColor))
+    }
+    return props.textColor ? getColor(props.textColor as string) : getColor(getTextColor(getColor(unref(componentColor))))
+  })
```

Comment:
Minor. Having two additional variables make code easier to read.
```suggestion
  const textColorComputed = computed(() => {
    const hexComponentColor = getColor(unref(componentColor))
    const textColorProp = props.textColor as string | undefined
  
    if (unref(isTransparent)) {
      return hexComponentColor
    }

    return getColor(textColorProp || getTextColor(hexComponentColor))
  })
```